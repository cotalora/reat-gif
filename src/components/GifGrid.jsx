import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);
    
    return (
        <>
            {
                isLoading && <h1>Cargando...</h1>
            }
            <h1>{category}</h1>
            <div className="card-grid">
                {
                    images?.map((image) => {
                        return (
                            <GifItem 
                                key={image.id}
                                {...image}
                            />
                        )
                    })
                }
            </div>
        </>
    )
}
