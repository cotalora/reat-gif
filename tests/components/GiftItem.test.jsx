import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas para componente GiftItem', () => {
    
    const url = 'https://www.google.com/saitama.jpg';
    const title = 'Saitama';
    
    test('Debería mostrar el compontente igual al match', () => {
        const { container } = render(<GifItem url={url} title={title}/>)
        expect(container).toMatchSnapshot();
    });
    
    test('Debería mostrar una imágen con URL y ALT', () => {
        render(<GifItem url={url} title={title}/>);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('Debería mostrar un texto en el componente', () => {
        render(<GifItem url={url} title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});