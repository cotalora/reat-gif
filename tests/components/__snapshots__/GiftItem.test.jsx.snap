// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para componente GiftItem Debería  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://www.google.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;

exports[`Pruebas para componente GiftItem Debería mostrar el compontente igual al match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://www.google.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
